course/views.py

```diff
@@ -1164,16 +1185,27 @@ def grant_exception_stage_3(pctx, participation_id, flow_id, session_id):
                     kind=flow_rule_kind.access,
                     rule=new_access_rule)
                 fre_access.save()
+                exception_created = True
 
             # }}}
 
             new_access_rules_tag = form.cleaned_data.get("set_access_rules_tag")
             if new_access_rules_tag == NONE_SESSION_TAG:
                 new_access_rules_tag = None
 
+            session_access_rules_tag_changed = False
             if session.access_rules_tag != new_access_rules_tag:
                 session.access_rules_tag = new_access_rules_tag
                 session.save()
+                session_access_rules_tag_changed = True
+
+                if new_access_rules_tag is not None:
+                    msg = _("Access rules tag of the selected session "
+                            "updated to '%s'.") % new_access_rules_tag
+                else:
+                    msg = _("Removed access rules tag of the selected session.")
+
+                messages.add_message(pctx.request, messages.SUCCESS, msg)
```

Comment:
Why doesn't this set `exception_created` to `True`?

Reply:
Because it doen't guarentee an exception object was or will be created. I've post a discussion (#551 ) about this.